src/hooks/useScrollAttributes/index.js

```diff
@@ -20,21 +20,24 @@ export const getScrollAttributes = (e, contentScrollWidth) => {
   };
 };
 
-const useScrollAttributes = () => {
+const useScrollAttributes = (ref) => {
   const [isAtScrollStart, setIsAtScrollStart] = useState(true);
   const [isAtScrollEnd, setIsAtScrollEnd] = useState(false);
-  const contentEl = useRef(null);
 
   const onScroll = (e) => {
-    const { scrollStart, scrollEnd } = getScrollAttributes(
-      e,
-      contentEl.current.scrollWidth
-    );
+    const { scrollStart, scrollEnd } = getScrollAttributes(e, ref.current.scrollWidth);
     rafSchd(setIsAtScrollStart(scrollStart));
     rafSchd(setIsAtScrollEnd(scrollEnd));
```

Comment:
Would these two be better as a single state update to an object with these 2 bools instead of 2 useState calls? My thinking is unless rafSchd can ensure batching those updates together during reconciliation, React might unmount this hook after the first & raise a `Can't perform a React state update on an unmounted component` warning. 

Something like this can help to prevent that:
```
[isAtScrollPositions, setIsAtScrollPositions] = useState({ start: true, end: false });
...
const { scrollStart: start, scrollEnd: end } = getScrollAttributes(e, ref.current.scrollWidth);
rafSchd(setIsAtScrollEnd({ start, end }));
```

Another alternative would be to use `useReducer` instead of `useState` entirely (but doesn't seem necessary in this case).

Reply:
Thanks! (talked offline regarding this). I "might" rather see this break on us as this would create some bigger pattern ramifications across the board for cbi-site if this is actually an anti-pattern 😅